source/FreeRTOS_Sockets.c

```diff
@@ -1926,8 +1926,17 @@ BaseType_t vSocketBind( FreeRTOS_Socket_t * pxSocket,
         if( pxAddress == NULL )
         {
             pxAddress = &xAddress;
-            /* Put the port to zero to be assigned later. */
-            pxAddress->sin_port = 0U;
+            /* Clear the address: */
+            ( void ) memset( pxAddress, 0, sizeof( struct freertos_sockaddr ) );
+
+            if( pxSocket->bits.bIsIPv6 == pdFALSE_UNSIGNED )
```

Comment:
Sorry, my bad, should be `pdTRUE_UNSIGNED`.

Does this compile with all combinations of `ipconfigUSE_IPv4/6`?

Reply:
Oh i missed it, let me update it. Yes it compiles with all the combination. Thanks